app/src/main/java/org/mozilla/fenix/settings/logins/controller/SavedLoginsStorageController.kt

```diff
@@ -0,0 +1,200 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.logins.controller
+
+import android.content.Context
+import android.util.Log
+import androidx.navigation.NavController
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import mozilla.components.concept.storage.Login
+import mozilla.components.service.sync.logins.InvalidRecordException
+import mozilla.components.service.sync.logins.LoginsStorageException
+import mozilla.components.service.sync.logins.NoSuchRecordException
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.settings.logins.LoginsAction
+import org.mozilla.fenix.settings.logins.LoginsFragmentStore
+import org.mozilla.fenix.settings.logins.fragment.EditLoginFragmentDirections
+import org.mozilla.fenix.settings.logins.mapToSavedLogin
+
+/**
+ * Controller for all saved logins interactions with the password storage component
+ */
+open class SavedLoginsStorageController(
+    context: Context,
+    private val viewLifecycleScope: CoroutineScope,
+    private val navController: NavController,
+    private val loginsFragmentStore: LoginsFragmentStore
+) {
+    private val activity: HomeActivity = context as HomeActivity
```

Comment:
This is a flow taken from a number of other places in the app. I would not like to change it without cleaning up every other place as well. Maybe we can file an eng health ticket.

Reply:
https://github.com/mozilla-mobile/fenix/blob/f42465657575f94e1d02e0ab95ee56581d3d1676/app/src/main/java/org/mozilla/fenix/settings/advanced/DefaultLocaleSettingsController.kt#L19

https://github.com/mozilla-mobile/fenix/blob/f42465657575f94e1d02e0ab95ee56581d3d1676/app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkController.kt#L52

https://github.com/mozilla-mobile/fenix/blob/f42465657575f94e1d02e0ab95ee56581d3d1676/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarController.kt#L63

https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt#L34

Reply:
It is a common pattern that we can file a followup ticket, but in this case we really are only using the context to get the `passwordsStorage`, so instead of passing in the context, we can just pass in the `passwordsStorage` directly.

Reply:
Changed this to use context instead of HomeActivity, and filed a followup: https://github.com/mozilla-mobile/fenix/issues/12565

Reply:
I think we can still improve this further by changing it pass `passwordsStorage` directly since AFAICT that's all we use it for

Reply:
is it cool to pass a component around like that?

Reply:
Ya! See https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/collections/CollectionCreationController.kt#L76